eox_nelp/external_certificates/api/v1/views.py

```diff
@@ -0,0 +1,108 @@
+
+    def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        """Upsert external certificate.
+        Create or update an external certificate based on the provided certificate response data.
+        Also associate the certificate with the corresponding user and result notification.
+
+        ## Usage
+        ### **POST** eox-nelp/api/external-certificates/v1/upsert/
+        **Request Body Example**
+        ```json
+        {
+            "user_id": "2",
+            "course_id": "course-v1:test+CS304+2025_T3",
+            "certificate_response": {
+                "message": "Certificate created successfully",
+                "certificate_id": "TEST25YO59VV76QL",
+                "certificate_urls": {
+                    "en": "https://ce.nelc.gov.sa/storage/pdf/OTT-lms/TEST25YO59VV76QL-en-1740573557.pdf",
+                    "ar": "https://ce.nelc.gov.sa/storage/pdf/OTT-lms/TEST25YO59VV76QL-ar-1740573555.pdf"
+                }
+            }
+        }
+        ```
+        **Response Example**
+        ```json
+        {
+            "user": "1",
+            "course_overview": "course-v1:test+CS304+2025_T3",
+            "certificate_id": "TEST25YO59VV76QL",
+            "certificate_url_en": "https://ce.nelc.gov.sa/storage/pdf/OTT-lms/TEST25YO59VV76QL-en-1740573557.pdf",
+            "certificate_url_ar": "https://ce.nelc.gov.sa/storage/pdf/OTT-lms/TEST25YO59VV76QL-ar-1740573555.pdf",
+            "created_at": "2025-02-26T23:25:02.364189Z"
+        }
+        ```
+        """
+        required_keys = ["certificate_response", "user_id", "course_id"]
+        missing_keys = [key for key in required_keys if not request.data.get(key)]
+
+        if missing_keys:
+            return Response(
+                {"error": f"Missing required keys: {missing_keys}"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            user = User.objects.get(id=request.data["user_id"])
+            course_overview = CourseOverview.objects.get(id=request.data["course_id"])
+            external_certificate = ExternalCertificate.create_external_certificate_from_certificate_response(
+                certificate_response=request.data["certificate_response"],
+                user=user,
+                course_overview=course_overview,
+            )
+        except User.DoesNotExist:
+            return Response(
+                {
+                    "error": f'user_id={request.data["user_id"]} User object with that id doesnt exists'
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        except CourseOverview.DoesNotExist:
+            return Response(
+                {
+                    "error": f'course_id={request.data["course_id"]} CourseOverview object with that id doesnt exists'
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        if not external_certificate:
```

Comment:
could you move this to the beginning and do something like this 
 ```suggestion
        if error in request.data["certificate_response"]:
```
As I mentioned in a different pr the validation could be performed before calling the method

Reply:
[04ea1e9](https://github.com/eduNEXT/eox-nelp/pull/268/commits/04ea1e9759d8de52f3dae394a9935ab2ae0cd0e2)